internal/hashing/hashing_test.go

```diff
@@ -20,65 +20,54 @@ func TestNew(t *testing.T) {
-	input := [][]byte{[]byte("transaction1"), []byte("transaction2")}
-	concat := append(New(New(input[0])), New(New(input[1]))...)
-	expected := New(New(concat))
-	actual := GetMerkleRootHash(input)
+func TestGetMerkleRootHashInput(t *testing.T) {
+	tests := []struct {
+		name string
+		input [][]byte
+		expected []byte
+		want bool
+	}{
+		{
+			"Empty Slice",
+			[][]byte{},
+			nil,
+			true,
+		},
+		{
+			"Size One Slice",
+			[][]byte{[]byte("transaction")},
+			New(New([]byte("transaction"))),
+			true,
+		},
+		{
+			"Size Two Slice",
+			[][]byte{[]byte("transaction1"), []byte("transaction2")},
+			New(New(append(New(New([]byte("transaction1"))), New(New([]byte("transaction2")))...))),
+			true,
 
-	if !bytes.Equal(expected, actual) {
-		t.Errorf("Error! GetMerkelRootHash does not produce correct result on two byte slices")
-		t.Errorf("Expected != Actual")
-		t.Errorf("%v != %v", expected, actual)
-	}
-}
+		},
+		{
+			"Size Three Slice",
+			[][]byte{[]byte("transaction1"), []byte("transaction2"), []byte("transaction3")},
+			New(New(append(New(New(append(New(New([]byte("transaction1"))), New(New([]byte("transaction2")))...))), New(New(append(New(New([]byte("transaction3"))), New(New([]byte("transaction3")))...)))...))),
+			true,
 
-func TestGetMerkleRootHashTripleInput(t *testing.T) {
-	input := [][]byte{[]byte("transaction1"), []byte("transaction2"), []byte("transaction3")}
-	concat1 := New(New(append(New(New(input[0])), New(New(input[1]))...)))
-	concat2 := New(New(append(New(New(input[2])), New(New(input[2]))...)))
-	expected := New(New(append(concat1, concat2...)))
-	actual := GetMerkleRootHash(input)
+		},
+		{
+			"Size Four Slice",
+			[][]byte{[]byte("transaction1"), []byte("transaction2"), []byte("transaction3"), []byte("transaction4")},
+			New(New(append(New(New(append(New(New([]byte("transaction1"))), New(New([]byte("transaction2")))...))), New(New(append(New(New([]byte("transaction3"))), New(New([]byte("transaction4")))...)))...))),
+			true,
 
-	if !bytes.Equal(expected, actual) {
-		t.Errorf("Error! GetMerkelRootHash does not produce correct result on three byte slices")
-		t.Errorf("Expected != Actual")
-		t.Errorf("%v != %v", expected, actual)
+		},
 	}
-}
 
-func TestGetMerkleRootHashQuadInput(t *testing.T) {
-	input := [][]byte{[]byte("transaction1"), []byte("transaction2"), []byte("transaction3"), []byte("transaction4")}
-	concat1 := New(New(append(New(New(input[0])), New(New(input[1]))...)))
-	concat2 := New(New(append(New(New(input[2])), New(New(input[3]))...)))
-	expected := New(New(append(concat1, concat2...)))
-	actual := GetMerkleRootHash(input)
-
-	if !bytes.Equal(expected, actual) {
-		t.Errorf("Error! GetMerkelRootHash does not produce correct result on three byte slices")
-		t.Errorf("Expected != Actual")
-		t.Errorf("%v != %v", expected, actual)
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if result := bytes.Equal(GetMerkleRootHash(tt.input),tt.expected); result != tt.want {
```

Comment:
can we just make the assertion fail if GetMerkleRootHash bytes do not equal expected?